packages/core/src/components/ui/ProductGallery/ProductGalleryPage.tsx

```diff
@@ -4,17 +4,33 @@ import ProductGrid from 'src/components/product/ProductGrid'
 import Sentinel from 'src/sdk/search/Sentinel'
 
 import { ProductCardProps } from 'src/components/product/ProductCard'
-import { useProducts } from './usePageProducts'
+import { memo, useMemo } from 'react'
+import { usePageProductsQuery } from 'src/sdk/product/usePageProductsQuery'
 
 interface Props {
   page: number
   title: string
   productCard?: Pick<ProductCardProps, 'showDiscountBadge' | 'bordered'>
+  itemsPerPage: number
 }
 
-function GalleryPage({ page, title, productCard }: Props) {
-  const products = useProducts(page) ?? []
-  const { itemsPerPage } = useSearch()
+function ProductGalleryPage({ page, title, productCard, itemsPerPage }: Props) {
+  const {
+    state: { term, sort, selectedFacets },
+  } = useSearch()
+
+  const { currentProductsPerPage } = usePageProductsQuery({
+    page,
+    term,
+    sort,
+    selectedFacets,
+    itemsPerPage,
+  })
+
+  const products = useMemo(
+    () => currentProductsPerPage?.data?.search?.products?.edges ?? [],
+    [currentProductsPerPage?.data?.search?.products?.edges]
```

Comment:
Is this `useMemo` really useful? Are we preventing rerenders because of this? Why?

Reply:
I thought we'd prevent rerenders from the children components. But in my tests now the `ProductGrid` wasn't rerendering for the sake of this memo. So I removed [here](https://github.com/vtex/faststore/pull/1967/commits/dc2ed8e405e54fee09246d797328aaac3b2fa795).